presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryProgressEvent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.eventlistener;
+
+import com.facebook.presto.common.resourceGroups.QueryType;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class QueryProgressEvent
```

Comment:
I think what makes this different from similar classes in this package is this is designed to publish multiple events, instead of an event per state transition or at query start/finish.  If you imagine that someone might put this into a queue that has at least once semantics, then it might be possible for messages to arrive out of order or duplicated.  That's not a problem for peer classes in this package because you know the state transition order, or you know that you only need to look at one query completed even if there's a duplicate, but for this class, it's impossible to infer correct ordering.  This may lead to strange queries or incorrect analysis of the events.  I'd recommend you include a monotonically increasing identifier that downstream systems can use to infer ordering and to de-duplicate at the reader side.

Reply:
@tdcmeehan : Added an increasing event Id field in the QueryProgressEvent class.